code/game/objects/effects/decals/cleanable/misc.dm

```diff
@@ -316,6 +316,26 @@
 	. = ..()
 	. += emissive_appearance(icon, "[icon_state]_light", alpha = src.alpha)
 
+/obj/effect/decal/cleanable/ants/fire_act(exposed_temperature, exposed_volume)
+	new /obj/effect/decal/cleanable/ants/fire(loc)
+	qdel(src)
```

Comment:
Why not make the fire ants take their reagent number from the original ant pile by running the reagent amount through this `new` proc into the fire ant's `Initialize()`?